### PR TITLE
STL-676 Add Seth RPC Block Calls

### PR DESCRIPTION
**Pull request by aludvik**
Implement block calls

Also do some other stuff that needs doin'

### CODE REVIEW

File: families/seth/rpc/src/block.rs [hunk 1]
```diff
@@ -60,9 +68,139 @@ pub fn block_number<T>(_params: Params, mut client: ValidatorClient<T>) -> Resul
...
+    let block_header: BlockHeader = match protobuf::parse_from_bytes(&block.header) {
+        Ok(r) => r,
+        Err(error) => {
+            error!("Error parsing block header: {:?}", error);
+            return Err(Error::internal_error());
+        },
+    };
+
+    // let mut bob: HashMap<String, Value> = HashMap::new();
+    // let mut bob: Value::Map<String, Value> = HashMap::new();
```
Comment (aludvik) [inline families/seth/rpc/src/block.rs]: WHAT ARE YOU DOING ADAM DON'T COMMIT COMMENTED CODE

File: protos/client.proto [hunk 2]
```diff
@@ -294,6 +294,7 @@ message ClientBlockListResponse {
 // specified by its unique id, in this case the block's header signature
 message ClientBlockGetRequest {
     string block_id = 1;
+    uint64 block_num = 2;
```
Comment (vaporos) [inline protos/client.proto]: We might want to do this in two separate requests for 1.0.  (Topic for Oct.)

File: families/seth/rpc/tests/test_seth_rpc.py [hunk 3]
```diff
@@ -63,3 +73,86 @@ def test_block_number(self):
                 )]),
             msg)
         self.assertEqual("0xf", self.rpc.get_result())
+
+    @unittest.skip("")
```
Comment (aludvik) [inline families/seth/rpc/tests/test_seth_rpc.py]: gaahhh

File: families/seth/rpc/src/requests.rs [hunk 4]
```diff
@@ -98,4 +108,55 @@ impl<S: MessageSender> ValidatorClient<S> {
...
+            ClientReceiptGetResponse_Status::OK => response.receipts,
+            ClientReceiptGetResponse_Status::INTERNAL_ERROR => {
+                return Err(String::from("Received internal error from validator"));
+            },
+            ClientReceiptGetResponse_Status::NO_RESOURCE => {
+                return Err(String::from("Missing receipt"));
+            }
+        };
+        let mut seth_receipts: HashMap<String, SethTransactionReceipt> = HashMap::new();
+        for rcpt in receipts.iter() {
```
Comment (peterschwarz) [inline families/seth/rpc/src/requests.rs]: You could consider using the Iterator function here, like `flat_map`, `filter`, `map` and `fold`

File: families/seth/rpc/src/block.rs [hunk 5]
```diff
@@ -60,9 +68,139 @@ pub fn block_number<T>(_params: Params, mut client: ValidatorClient<T>) -> Resul
...
+    let mut transactions = Vec::new();
+    let mut gas: u64 = 0;
+    for (txn_id, receipt) in receipts.into_iter() {
+        transactions.push(string_to_hex(&txn_id));
+        gas += receipt.gas_used;
+    }
+    bob.insert(String::from("transactions"), Value::Array(transactions));
+    bob.insert(String::from("gasUsed"), num_to_hex(&gas));
+
+    // No correlary in Sawtooth
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply families/seth/rpc/src/block.rs]: corollaries

**Pull request merged**